laufband/laufband.py

```diff
@@ -110,50 +98,66 @@ def __init__(
         ...        output_file.write_text(json.dumps(file_content))
 
         """
-        if heartbeat_timeout is None:
-            heartbeat_timeout = int(os.getenv("LAUFBAND_HEARTBEAT_TIMEOUT", 60 * 60))
-        if max_died_retries is None:
-            max_died_retries = int(os.getenv("LAUFBAND_MAX_DIED_RETRIES", 0))
-        if identifier is None:
-            identifier = os.getenv("LAUFBAND_IDENTIFIER", str(os.getpid()))
-        if lock_path is not None and lock is not None:
-            raise ValueError(
-                "You cannot set both `lock` and `lock_path`. Use one or the other."
-            )
-        if disable is None:
-            self.disabled = os.getenv("LAUFBAND_DISABLE", "0") == "1"
-        else:
-            self.disabled = disable
+        self.data = data
 
-        if lock_path is None:
-            lock_path = "laufband.lock"
+        # Store the data source for lazy evaluation
+        self._data_source = data
+        self._item_mapping = {}
+        self._mapping_created = False
+
+        # Create a graph_fn that returns a graph with disconnected nodes (no edges)
+        def graph_fn():
+            # Lazy evaluation - create mapping only when graph is requested
+            if not self._mapping_created:
+                import uuid
+
+                for item in self._data_source:
+                    item_uuid = str(uuid.uuid4())
+                    self._item_mapping[item_uuid] = item
+                self._mapping_created = True
+
+            G = nx.DiGraph()
+            for item_uuid in self._item_mapping.keys():
+                G.add_node(item_uuid)
+                # Store the actual item as node data for access
+                G.nodes[item_uuid]["value"] = self._item_mapping[item_uuid]
+            return G
```

Comment:
I don't think this is, what I want? Here, again we are iterating the generator whole to construct the graph. We would need `graph.from_generator` that can then be consumed by the search but that probably doesn't exist. Also graph_fn returns a graph and itself is not a generator.

Reply:
This could be fixed, by just iterating until we find an item that has not yet been processed, but that doesn't really work. We could just yield nodes / edges and thus the user could define how to iterate through the data - default would be topological sort / bfs?

Reply:
The original idea was, to keep the generator at the given state and when the next item is requested, move on until an unprocessed item is found. This is not compatible with inserting new items but appending would work. Currently, we allow for insertion but this is not tested?